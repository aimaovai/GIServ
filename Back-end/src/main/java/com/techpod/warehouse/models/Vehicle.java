package com.techpod.warehouse.models;

import javax.persistence.*;

@Entity
@Table(name = "vehicles")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    public String name;

    public int number;

    public boolean free;
}
