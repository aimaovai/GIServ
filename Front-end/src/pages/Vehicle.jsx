import React, { useEffect, useState } from "react";
import api from "../api";
import Layout from "../components/Layout";
import handleForm from "../helper";
import { checkLoggedIn } from "../helper/AuthHelper";

export default function Vehicle() {
  const [Vehicles, setVehicles] = useState(null);
  const [form, setForm] = useState({
    name: "",
    number: "",
    free: true,
  });

  const [stats, setStats] = useState({
    total: 0,
    free: 0,
    notfree: 0,
  });

  useEffect(() => {
    checkLoggedIn();    

    getAllVehicles();
  }, []);

  const newVehicle = () => {

    if(form.name === "" || form.number === ""){
      alert("Fields Cannot Be Empty")
        return;
    }

    console.log(form);
    api.addVehicle(form).then((res) => {
      getAllVehicles();
      alert("Successfully Vehicle Added");
      setForm({
        name: "",
        number: "",
        free: true,
      });
    });
  };

  const getAllVehicles = async () => {
    let Vehicles = await api.getVehicles();
    console.log(Vehicles);

    let free = 0,
      notfree = 0;

    Vehicles.map((vehicle) => {
      vehicle.free ? (free += 1) : (notfree += 1);
    });

    setStats({
      total: Vehicles.length,
      free,
      notfree,
    });
    setVehicles(Vehicles);
  };

  const removeVehicle = (id) => {
    api.deleteVehicle(id).then((res) => {
      getAllVehicles();
      alert("Successfully Vehicle Removed");
    });
  };

  const update = (data) => {
    let tempData = {
      ...data,
      free: !data.free,
    };
    api.addVehicle(tempData).then((res) => {
      getAllVehicles();
    });
  };

  return (
    <Layout>
      <div className="flex justify-around">
        <div className="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2">
          <p className="text-4xl">{stats.total}</p>
          <p className="text-sm">Total</p>
        </div>
        <div className="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2">
          <p className="text-4xl">{stats.free}</p>
          <p className="text-sm">Available</p>
        </div>
        <div className="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2">
          <p className="text-4xl">{stats.notfree}</p>
          <p className="text-sm">Not Available</p>
        </div>
      </div>


      <div data-testid="vehinsert" className="flex gap-2 pt-5 pb-2 justify-center">
        <input
          type="text"
          placeholder="name"
          name="name"
          value={form.name}
          className="h-10 outline-none bg-slate-100 text-center rounded-md"
          onChange={(e) => setForm(handleForm(e, form))}
        />
        <input
          type="text"
          placeholder="number"
          name="number"
          value={form.number}
          className="h-10 outline-none bg-slate-100 text-center rounded-md"
          onChange={(e) => setForm(handleForm(e, form))}
        />
        <button
          data-testid="vinsert-button"
          className="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          onClick={() => newVehicle()}
        >
          Add Vehicle
        </button>
      </div>
      <table data-testid="vehicles" className="w-full text-center my-2 rounded-md bg-slate-50 ">
        <thead>
          <tr className="">
            <th>Id</th>
            <th>Name</th>
            <th>Number</th>
            <th>Available</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {Vehicles &&
            Vehicles.map((Vehicle) => {
              return (
                <tr className="py-3">
                  <td>{Vehicle.id}</td>
                  <td>{Vehicle.name}</td>
                  <td>{Vehicle.number}</td>
                  <td>
                    {Vehicle.free ? "Yes" : "No"}
                    <button
                      data-testid="toggle-button"
                      onClick={() => update(Vehicle)}
                      className="text-sm bg-green-300 p-1 rounded-md m-1"
                    >
                      Change
                    </button>
                  </td>
                  <td>
                    <button
                      className="text-sm bg-red-50 text-red-500 px-2 py-1 rounded-md"
                      onClick={() => removeVehicle(Vehicle.id)}
                    >
                      Remove
                    </button>
                  </td>
                </tr>
              );
            })}
        </tbody>
      </table>
    </Layout>
  );
}
