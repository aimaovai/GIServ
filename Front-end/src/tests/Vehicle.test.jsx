/*
  - Test suite id: TS_06
  - Test suite written by: Aima Ovai
  - Test suite written on: April 7, 2022
  - Test suite last updated on: April 28, 2022
  - Test Description: This test suite was created to confirm that the Vehicles
    webpage renders properly and without crashing. It also contains test cases
    that check that the data for the vehicles are being retrieved and are displayed
    in the table.
*/

import React from "react";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, mount, configure } from "enzyme";
import { render, screen, waitFor,fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Vehicle from '../pages/Vehicle';

// jest.mock('../helper/AuthHelper');

// const func = require('../helper/AuthHelper');

Enzyme.configure({adapter: new Adapter()});

describe('GIServ Vehicle Component', () => {

  // Test case: TC_18
  it('user is logged in', () => {
    try {
      const loggedin = jest.fn().mockReturnValueOnce("loggedin");
      Storage.prototype.setItem = jest.fn(() => {
        console.log("isUserLoggedin")
      });

      checkLoggedIn.setItem("isUserLoggedin", loggedin);
      } catch (error) {
        console.log('User not logged in.')
      }
  })
  
  // Test case: TC_19
  it("Vehicle renders without crashing", async () => {
    render(<Vehicle />)

    // await expect(render(<Vehicle />)).toMatchSnapshot()
    const vehiclelItem = await waitFor(() => screen.getByTestId('vehicles'));
      expect(vehiclelItem).toBeInTheDocument();
  });

  // Test case: TC_20
  test('it displays a table for Vehicles', async () => {
    
      render(<Vehicle />);

      const vehicleList = await waitFor(() => screen.getByTestId('vehicles'));
      expect(vehicleList).toBeInTheDocument();
  })

  // Test case: TC_21
  test('it retrieves data', async () => {
   
      render(<Vehicle />);

      const vehicleList = await waitFor(() => screen.findAllByTestId('vehicles'));
      expect(vehicleList).toHaveLength(1);
  })

  // Test case: TC_22
  test('test Vehicle insert button on click', () => {
    const { getByTestId } = render(<Vehicle />);
    fireEvent.click(getByTestId('vinsert-button'));
    expect(getByTestId('vehinsert')).toHaveTextContent('Add Vehicle');
  });

  // Test case: TC_23
  test('test that Vehicle status toggle button exists', () => {

      const wrapper = mount(<Vehicle />);
      const table = wrapper.find('table');
      const row = table.find('tr');
      const togleButton = table.find('toggle-button');

      expect(table).toHaveLength(1);
      expect(row).toHaveLength(1);
      expect(togleButton).not.toBeNull();
  });
})