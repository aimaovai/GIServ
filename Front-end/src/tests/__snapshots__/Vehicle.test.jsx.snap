// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIServ Vehicle Component Vehicle renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
      >
        <div
          class="flex justify-around"
        >
          <div
            class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Total
            </p>
          </div>
          <div
            class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Available
            </p>
          </div>
          <div
            class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Not Available
            </p>
          </div>
        </div>
        <div
          class="flex gap-2 pt-5 pb-2 justify-center"
        >
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="name"
            placeholder="name"
            type="text"
            value=""
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="number"
            placeholder="number"
            type="text"
            value=""
          />
          <button
            class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          >
            Add Vehicle
          </button>
        </div>
        <table
          class="w-full text-center my-2 rounded-md bg-slate-50 "
        >
          <thead>
            <tr
              class=""
            >
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Number
              </th>
              <th>
                Available
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
    <div>
      <div
        class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
      >
        <div
          class="flex justify-around"
        >
          <div
            class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Total
            </p>
          </div>
          <div
            class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Available
            </p>
          </div>
          <div
            class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Not Available
            </p>
          </div>
        </div>
        <div
          class="flex gap-2 pt-5 pb-2 justify-center"
        >
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="name"
            placeholder="name"
            type="text"
            value=""
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="number"
            placeholder="number"
            type="text"
            value=""
          />
          <button
            class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          >
            Add Vehicle
          </button>
        </div>
        <table
          class="w-full text-center my-2 rounded-md bg-slate-50 "
        >
          <thead>
            <tr
              class=""
            >
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Number
              </th>
              <th>
                Available
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
    >
      <div
        class="flex justify-around"
      >
        <div
          class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Total
          </p>
        </div>
        <div
          class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Available
          </p>
        </div>
        <div
          class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Not Available
          </p>
        </div>
      </div>
      <div
        class="flex gap-2 pt-5 pb-2 justify-center"
      >
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          name="name"
          placeholder="name"
          type="text"
          value=""
        />
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          name="number"
          placeholder="number"
          type="text"
          value=""
        />
        <button
          class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
        >
          Add Vehicle
        </button>
      </div>
      <table
        class="w-full text-center my-2 rounded-md bg-slate-50 "
      >
        <thead>
          <tr
            class=""
          >
            <th>
              Id
            </th>
            <th>
              Name
            </th>
            <th>
              Number
            </th>
            <th>
              Available
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
